lib/joystick.dart

```diff
@@ -0,0 +1,125 @@
+import 'dart:math';
+import 'dart:ui';
+
+import 'package:flame/anchor.dart';
+import 'package:flame/components/component.dart';
+
+class Joystick extends Component {
+  double _movementRadius = 200;
+  double _radiusSquare = 40000;
+  Offset _offset = Offset.zero;
+
+  /// The base part of the joystick. It's like the area in which the stick swivels.
+  ///
+  /// You shouldn't change it's position manually, but rather use the Joystick methods.
+  final SpriteComponent base = SpriteComponent()..anchor = Anchor.center;
```

Comment:
I am not sure by the way `SpriteComponent` is used here.

All components are supposed to be added to a base game class, and not be handled (update and render) manually.

Since you are handling it manually here, I think you should use the plain `Sprite` class, or add these `SpriteComponent` to the game, you just need to make sure that their destroy methods are tied to this component.